devlib/target.py

```diff
@@ -279,6 +280,11 @@ def shutils(self):
             self._setup_shutils()
         return self._shutils
 
+    def is_running(self, process_regex):
+        cmd = f"{self.busybox} ps -A -o comm,stat | {self.busybox} awk '$1 ~ /{process_regex}/ && $2 !~ /Z/'b"
```

Comment:
I'd be _very_ wary of introducing a regex here. We had done that in the freezer cgroup with a grep regex function and that is a great recipe for maintenance nightmare: the second you add that, you tied the implementation of that function to awk regex, forever.

If we really need that feature, I would heavily recommend getting the whole list of PID, and applying the regex on the Python side, using a Python regex, which will not bake any implementation detail in the API. On top of that, that would mean the devlib API would now get 3 flavors of regex (Python, awk, grep), which clearly is not great.